src/api/app/models/workflow/step/link_package_step.rb

```diff
@@ -0,0 +1,102 @@
+class Workflow::Step::LinkPackageStep < ::Workflow::Step
+  validates :source_package_name, presence: true
+
+  def call(options = {})
+    return unless valid?
+
+    linked_package = find_or_create_linked_package
+    add_or_update_branch_request_file(package: linked_package)
+    workflow_filters = options.fetch(:workflow_filters, {})
+    create_or_update_subscriptions(linked_package, workflow_filters)
+
+    workflow_repositories(target_project_name, workflow_filters).each do |repository|
+      # TODO: Fix n+1 queries
```

Comment:
are we going to fix the n+1 queries on this PR or later?

Reply:
We should fix that for both `link_package` and `branch_package` steps. But we don't plan to do it now.